cirq-google/cirq_google/calibration/workflow.py

```diff
@@ -50,7 +50,12 @@
 from cirq_google.engine import Engine, QuantumEngineSampler
 from cirq_google.serialization.serializer import Serializer
 
-_CALIBRATION_IRRELEVANT_GATES = cirq.MeasurementGate, cirq.SingleQubitGate, cirq.WaitGate
+_CALIBRATION_IRRELEVANT_GATES = (
+    cirq.MeasurementGate,
+    cirq.SingleQubitGate,
+    cirq.WaitGate,
+    cirq.GlobalPhaseGate,
```

Comment:
Add test to verify that `GlobalPhaseGate` is irrelevant for calibration and is correctly handled. I only see deletions for `cirq.GlobalPhaseOperation` in `workflow_test.py`.

Reply:
@mpharrigan Now with global phase being a gate instead of an op, how would/should this affect the calibration workflow? Previously it would just error out with IncompatibleMomentError because it was a gateless operation (you can see the deleted test portion below). With it being a gate now, it no longer raises that error. But is there any special handling or testing that needs to be done here?

Reply:
I added some new tests below based on my very naive understanding of what should be tested here. @mpharrigan can you review and make sure these make sense? Unless @tanujkhattar also understands these workflow requirements.

I based the logic on what we were doing for Z gates in the tests. But maybe this doesn't make sense since a global phase gate isn't a physical gate? (or is it?) Another option is just to have the workflow code reject global phase gates explicitly, like it does for gateless operations.

Reply:
I fixed it so it behaves exactly as before -- global phase throws an IncompatibleMomentError.

If there's some way we can make use of a global phase now that it has a gate representation, then that can be done separately by the maintainers of this functionality.